pkg/gql/httpserver.go

```diff
@@ -90,8 +96,50 @@ func (s *Server) Start() error {
 	s.schema = &sc
 	_, s.db = heavy.CreateDBConnection()
 
-	// Set up listener
-	l, err := net.Listen("tcp", "localhost:"+cfg.Get().Gql.Port)
+	if network == "unix" {
+		if err := os.RemoveAll(address); err != nil {
+			return err
+		}
+	}
+
+	var tlsCert *tls.Certificate
+
+	certFile := cfg.Get().Gql.CertFile
+	keyFile := cfg.Get().Gql.KeyFile
+
+	if len(certFile) > 0 {
+
+		// Fetch certificate from config
+		tlsCertFromFile, err := tls.LoadX509KeyPair(certFile, keyFile)
+		if err != nil {
+			return err
+		}
+		tlsCert = &tlsCertFromFile
+	} else {
+
+		// Generate self-signed certificate
+		tlsCert, err = cryptoUtils.GenerateTLSCertificate()
+		if err != nil {
+			return err
+		}
+	}
+
+	// Define TLS configuration
+	tlsConfig := tls.Config{
```

Comment:
Should we also specify the TLS min version, as by default it's TLS 1.0?

```
// MinVersion contains the minimum SSL/TLS version that is acceptable.
 // If zero, then TLS 1.0 is taken as the minimum.
 MinVersion uint16 // Go 1.2
```

Reply:
You are correct. the minimum required version for the voucher is 1.3

Reply:
Done